src/test/java/org/javarosa/xform/parse/ExternalSecondaryInstanceTest.java

```diff
@@ -0,0 +1,49 @@
+package org.javarosa.xform.parse;
+
+import org.javarosa.core.model.FormDef;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.model.condition.IFunctionHandler;
+import org.javarosa.core.test.FormParseInit;
+import org.junit.Test;
+
+import java.nio.file.Path;
+import java.util.List;
+
+import static org.javarosa.core.reference.ReferenceManagerTestUtils.setUpSimpleReferenceManager;
+import static org.javarosa.test.utils.ResourcePathHelper.r;
+
+public class ExternalSecondaryInstanceTest {
+    @Test
+    public void externalInstanceDeclaration_ShouldBeIgnored_WhenNotReferenced() {
+        Path formPath = r("minimal-pulldata.xml");
+        setUpSimpleReferenceManager("file-csv", formPath.getParent());
+        FormParseInit fpi = new FormParseInit(formPath);
+        FormDef formDef = fpi.getFormDef();
+        formDef.getEvaluationContext().addFunctionHandler(new IFunctionHandler() {
+            @Override
+            public String getName() {
+                return "pulldata";
+            }
+
+            @Override
+            public List<Class[]> getPrototypes() {
+                return null;
+            }
+
+            @Override
+            public boolean rawArgs() {
+                return false;
+            }
+
+            @Override
+            public boolean realTime() {
+                return false;
+            }
+
+            @Override
+            public Object eval(Object[] args, EvaluationContext ec) {
+                return null;
+            }
+        });
+    }
```

Comment:
So, I'll state the obvious: this test doesn't have assertions. 

We want to test that "this piece of code doesn't throw an exception". How could we assert that?